pkg/rbac/authenticationMiddleware_test.go

```diff
@@ -1,2 +1,137 @@
 // Copyright Contributors to the Open Cluster Management project
 package rbac
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+//test valid token from cookie
+func TestTokenCookieAuthenticated(t *testing.T) {
+
+	token := "validtoken"
+	authenticateHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+
+	r := httptest.NewRequest("GET", "https://localhost:4010/playground", nil)
+
+	r.AddCookie(&http.Cookie{Name: "acm-access-token-cookie", Value: token})
```

Comment:
You can add the token directly here, no need to declare it before. Same for the tests below.
```suggestion
	r.AddCookie(&http.Cookie{Name: "acm-access-token-cookie", Value: "mytesttoken"})
```